### DISCUSSION TITLE
open shapefile raised error: Assigning CRS to a GeoDataFrame without a geometry column is not supported...

### DISCUSSION THREAD

**Discussion by shuai-zhou** [geopandas/geopandas#2991] [General]
I created a new Python env (python=3.9) on a new Windows 10 machine, and installed geopandas 0.12.2
When trying to open a shapefile using `gdf = gpd.read_file('file_name.shp')`, I got an error, `ValueError: Assigning CRS to a GeoDataFrame without a geometry column is not supported. Supply geometry using the 'geometry=' keyword argument, or by providing a DataFrame with column name 'geometry'`. Why is this happening and how to fix it? Thank you.

**Comment by martinfleis** [comment]
That is a weird one. Can you

1. post the full traceback so we can track where this comes from?
2. try using the latest geopandas (0.13)
3. share the file?

**Reply by shuai-zhou** [reply]
This error occurred to every .shp/.zip file I tried to open, and I knew those files had a "geometry" column. For example, I downloaded a shapefile from the Natural Earth, you can download the file from [here](https://github.com/shuai-zhou/DataRepo/blob/master/data/ne_110m_admin_0_countries.zip). The traceback is as follows:

![image](https://github.com/geopandas/geopandas/assets/22311821/84d640af-27f2-455d-b027-a920a58df703)

Thank you for troubleshooting this issue.

**Reply by martinfleis** [reply]
Can you ensure you are using the latest `geopandas` (0.13) and `fiona` (1.9.4) before I'll do any debugging?

**Reply by shuai-zhou** [reply]
Never mind. I messed up the environment. I solved the issue by creating a new environment and strictly following the instruction provided by geopandas

`conda create -n geo_env`
`conda activate geo_env`
`conda config --env --add channels conda-forge`
`conda config --env --set channel_priority strict`
`conda install python=3 geopandas`
